`pytest` fails on CPU due to `SimCLRTrainerTest`

Our test suite does not run on CPU due to `SimCLRTrainerTest.test_train_without_probing`. Is this something we could fix to enable CPU-based development?

```sh
E     tensorflow.python.framework.errors_impl.InvalidArgumentError: {{function_node __wrapped__MakeIterator_device_/job:localhost/replica:0/task:0/device:CPU:0}} No OpK
ernel was registered to support Op 'AdjustContrastv2' used by {{node sim_clr_augmenter/maybe_apply_1/map/while/body/_273/sim_clr_augmenter/maybe_apply_1/map/while/cond/
then/_1048/sim_clr_augmenter/maybe_apply_1/map/while/cond/random_color_jitter/random_contrast/adjust_contrast}} with these attrs: [T=DT_HALF]
E     Registered devices: [CPU]
E     Registered kernels:
E       device='XLA_CPU_JIT'; T in [DT_FLOAT, DT_HALF]
E       device='CPU'; T in [DT_FLOAT]
E       device='GPU'; T in [DT_HALF]
E       device='GPU'; T in [DT_FLOAT]
E       device='XLA_GPU_JIT'; T in [DT_FLOAT, DT_HALF]
E     
E        [[sim_clr_augmenter/maybe_apply_1/map/while/body/_273/sim_clr_augmenter/maybe_apply_1/map/while/cond/then/_1048/sim_clr_augmenter/maybe_apply_1/map/while/cond/
random_color_jitter/random_contrast/adjust_contrast]] [Op:MakeIterator]

../../.pyenv/versions/3.10.0/envs/keras-cv/lib/python3.10/site-packages/tensorflow/python/framework/ops.py:7215: InvalidArgumentError
```

--- a/keras_cv/models/segmentation/deeplab_test.py
+++ b/keras_cv/models/segmentation/deeplab_test.py
@@ -77,6 +77,7 @@ class DeeplabTest(tf.test.TestCase):
         output = model(input_image, training=True)
 
         self.assertEquals(output["output"].dtype, tf.float32)
+        tf.keras.mixed_precision.set_global_policy("float32")
 
     def test_invalid_backbone_model(self):
         with self.assertRaisesRegex(

